src/tools/schema2graph.ts

```diff
@@ -184,9 +231,13 @@ export class SchemaGraph {
         nestedSchema = descriptor.schema.schema.model.entitySchema;
       }
       if (nestedSchema) {
+        // When a type variable has a nested schema, it should be backed by a) a distinct entity from
+        // a schema with the same name and b) a distinct entity from the original type variable.
+        // To accomplish this, we need to associate the nested schema with a "child" type variable.
+        const nestedVar = fromVariable ? fromVariable + field : '';
```

Comment:
maybe:
const nestedVar = fromVariable && '${fromVariable}.${field}';

2 things: handle `null` as suggested above and I like separating strings with `.` so that there's no risk of conflict.

Reply:
👍